cpp/src/arrow/compute/kernels/scalar_arithmetic_benchmark.cc

```diff
@@ -67,89 +68,35 @@ static void ArrayArrayKernel(benchmark::State& state) {
   for (auto _ : state) {
     ABORT_NOT_OK(Op(lhs, rhs, nullptr).status());
   }
+  state.SetItemsProcessed(state.iterations() * array_size);
 }
 
 void SetArgs(benchmark::internal::Benchmark* bench) {
-  bench->Unit(benchmark::kMicrosecond);
-
-  for (const auto size : kMemorySizes) {
+  for (const auto size : {kL1Size, kL2Size}) {
```

Comment:
I have a processor with 22MB L3 cache so generating that much random data is quite expensive. If we want to benchmark arrays that big we should generate a smaller sample of random data and repeat/tile it to make the bigger array.